les/clientpool.go

```diff
@@ -460,44 +581,53 @@ func (f *clientPool) addBalance(id enode.ID, amount uint64, setTotal bool) {
 			c.balanceTracker.setBalance(pb.value, negBalance)
 			if !c.priority && pb.value > 0 {
 				c.priority = true
+				if c.priority {
+					f.priorityConnected += c.capacity
+				}
 				c.balanceTracker.addCallback(balanceCallbackZero, 0, func() { f.balanceExhausted(id) })
 			}
+			c.balanceMetaInfo = meta
```

Comment:
In this defer function, even the error is returned, meta of `clientInfo` will be updated.

Reply:
fixed